testing/ef-tests/src/macros/fork_choice.rs

```diff
@@ -0,0 +1,256 @@
+                    pub block: String,
+                    pub blobs: Option<String>,
+                    pub proofs: Option<Vec<String>>,
+                    pub valid: Option<bool>,
+                }
+
+                #[derive(Debug, Deserialize)]
+                pub struct AttestationStep {
+                    pub attestation: String,
+                    #[serde(default)]
+                    pub valid: Option<bool>,
+                }
+
+                #[derive(Debug, Deserialize)]
+                pub struct AttesterSlashingStep {
+                    pub attester_slashing: String,
+                    #[serde(default)]
+                    pub valid: Option<bool>,
+                }
+
+                #[derive(Deserialize)]
+                #[serde(untagged)]
+                pub enum ForkChoiceStep {
+                    Tick(Tick),
+                    Checks { checks: Checks },
+                    Block(Block),
+                    Attestation(AttestationStep),
+                    AttesterSlashing(AttesterSlashingStep),
+                }
+
+                #[tokio::test]
+                async fn test_fork_choice() {
+                    let base_path = format!(
+                        "mainnet/tests/mainnet/deneb/fork_choice/{}/pyspec_tests",
+                        stringify!($path)
+                    );
+
+                    let mock_engine = MockExecutionEngine::new();
+
+                    for entry in std::fs::read_dir(base_path).unwrap() {
+                        let entry = entry.unwrap();
+                        let case_dir = entry.path();
+
+                        if !case_dir.is_dir() {
+                            continue;
+                        }
+
+                        let case_name = case_dir.file_name().unwrap().to_str().unwrap();
+                        println!("Testing case: {}", case_name);
+
+                        let steps: Vec<ForkChoiceStep> = {
+                            let steps_path = case_dir.join("steps.yaml");
+                            let content =
+                                std::fs::read_to_string(&steps_path).expect("Failed to read steps.yaml");
+                            serde_yaml::from_str::<Vec<ForkChoiceStep>>(&content)
+                                .expect("Failed to parse steps.yaml")
+                        };
+
+                        let anchor_state: BeaconState =
+                            utils::read_ssz_snappy(&case_dir.join("anchor_state.ssz_snappy"))
+                                .expect("Failed to read anchor_state.ssz_snappy");
+                        let anchor_block: BeaconBlock =
+                            utils::read_ssz_snappy(&case_dir.join("anchor_block.ssz_snappy"))
+                                .expect("Failed to read anchor_block.ssz_snappy");
+
+                        let mut store = get_forkchoice_store(anchor_state.clone(), anchor_block.clone())
+                            .expect("get_forkchoice_store failed");
```

Comment:
```suggestion

                        let mut store = get_forkchoice_store(anchor_state, anchor_block)
                            .expect("get_forkchoice_store failed");
```

We don't need this clone